src/composables/canvas/useSelectionToolboxPosition.ts

```diff
@@ -168,50 +168,58 @@ export function useSelectionToolboxPosition(
     }
   )
 
-  // Watch for dragging state
-  watch(
-    () => canvasStore.canvas?.state?.draggingItems,
-    (dragging) => {
-      if (dragging) {
-        visible.value = false
-
-        if (moreOptionsOpen.value) {
-          const currentSig = buildSelectionSignature(canvasStore)
-          if (currentSig !== moreOptionsSelectionSignature) {
-            moreOptionsSelectionSignature = null
-          }
-          moreOptionsWasOpenBeforeDrag = true
-          moreOptionsOpen.value = false
-          moreOptionsRestorePending.value = !!moreOptionsSelectionSignature
-          if (moreOptionsRestorePending.value) {
-            forceCloseMoreOptionsSignal.value++
-          } else {
-            moreOptionsWasOpenBeforeDrag = false
-          }
+  // Helper function to handle drag state changes (shared between LiteGraph and Vue nodes)
+  const handleDragStateChange = (dragging: boolean) => {
+    if (dragging) {
```

Comment:
Could you refactor this to flatten it out?
Ideally (but maybe not feasibly) there would be a single point of assignment for each of the refs.
At the very least, guard clauses could help reduce the `else`s.